test/test_config_parser.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+require 'test/unit'
+require 'config_parser'
+require 'config_pb'
+require 'fluent/config/v1_parser'
+
+# tests for config tool
+class TestConfigParser < Test::Unit::TestCase
+  # test to ensure comments are not parsed
+  def test_comments
+    dir1 = Config::Directive.new(name: 'system', params: [get_param('rpc', '0.0.0:2')])
+    dir2 = Config::Directive.new(name: 'source', params: [get_param('@type', 'forward')])
+    comment = Config::Directive.new(name: 'ROOT', directives: [dir1, dir2])
```

Comment:
QQ: Is it possible to just inline the sub-directives?  For example:

    expected = Config::Directive.new(
      name: 'ROOT',
      directives: [
        Config::Directive.new(name: 'system', params: [get_param('rpc', '0.0.0:2')]),
        Config::Directive.new(name: 'source', params: [get_param('@type', 'forward')])
      ]
    )

Reply:
This is a great suggestion - I will try to incorporate this as much as possible!